followthemoney/model.py

```diff
@@ -62,49 +64,59 @@ def get_type_schemata(self, type_):
+    def make_mapping(self, mapping: Dict[str, Any], key_prefix=None):
         """Parse a mapping that applies (tabular) source data to the model."""
         return QueryMapping(self, mapping, key_prefix=key_prefix)
 
-    def map_entities(self, mapping, key_prefix=None):
+    def map_entities(self, mapping: Dict[str, Any], key_prefix=None):
         """Given a mapping, yield a series of entities from the data source."""
-        mapping = self.make_mapping(mapping, key_prefix=key_prefix)
-        for record in mapping.source.records:
-            for entity in mapping.map(record).values():
+        q_mapping = self.make_mapping(mapping, key_prefix=key_prefix)
+        for record in q_mapping.source.records:
+            for entity in q_mapping.map(record).values():
                 yield entity
 
-    def common_schema(self, left, right):
+    def common_schema(self, left: Union[str, Schema], right: Union[str, Schema]) -> Schema:
         """Select the most narrow of two schemata.
 
         When indexing data from a dataset, an entity may be declared as a
         LegalEntity in one query, and as a Person in another. This function
         will select the most specific of two schemata offered. In the example,
         that would be Person.
         """
-        left = self.get(left) or self.get(right)
-        right = self.get(right) or self.get(left)
-        if left.is_a(right):
-            return left
-        if right.is_a(left):
-            return right
-        common = list(left.schemata.intersection(right.schemata))
+        left_schema = self.get(left) or self.get(right)
+        if left_schema is None:
+            # neither left nor right are valid schemata
+            msg = "Unknown schemata: %s and %s"
+            raise InvalidData(msg % (left, right))
+
+        right_schema = self.get(right) or self.get(left)
+        if right_schema is None:
+            # neither left nor right are valid schemata
+            msg = "Unknown schemata: %s and %s"
+            raise InvalidData(msg % (left, right))
+
+        if left_schema.is_a(right_schema):
+            return left_schema
+        if right_schema.is_a(left_schema):
+            return right_schema
+        common = list(left_schema.schemata.intersection(right_schema.schemata))
         if not len(common):
             msg = "No common ancestor: %s and %s"
-            raise InvalidData(msg % (left, right))
+            raise InvalidData(msg % (left_schema, right_schema))
 
         specific = common[0]
         for schema in common[1:]:
             if schema.is_a(specific):
                 specific = schema
         return specific
 
-    def make_entity(self, schema, key_prefix=None):
+    def make_entity(self, schema: Mapping, key_prefix=None) -> EntityProxy:
         return EntityProxy(self, {'schema': schema}, key_prefix=key_prefix)
 
-    def get_proxy(self, data, cleaned=True):
+    def get_proxy(self, data: Mapping, cleaned=True) -> EntityProxy:
```

Comment:
Union[Mapping, EntityProxy]

Reply:
For that, `EntityProxy` need to inherit `Mapping`